core/src/main/java/com/graphhopper/routing/util/AreaIndex.java

```diff
@@ -56,18 +57,28 @@ public AreaIndex(List<T> areas) {
     }
 
     private void addBorder(PreparedGeometryFactory pgf, T area, Polygon border) {
-        // todonow: do we really need the prepared geometry for osm areas?
         IndexedCustomArea<T> indexedCustomArea = new IndexedCustomArea<>(area, pgf.create(border));
         index.insert(border.getEnvelopeInternal(), indexedCustomArea);
```

Comment:
Did you try anything in this direction?

Reply:
No, I didn't. I'm also not sure what you mean by that comment.

Reply:
Sorry, I didn't want to delete your comment. I restored it now.

Reply:
Ok, I tried this in the meantime. The SRTRee only allows querying for rectangular shapes, so use it for the envelopes of the areas. But once we query the index and get the envelopes we need a more precise intersection check and this would be too slow if we did not create a prepared geometry for each area (I tested this and it was about 10x slower). So we need to prepare the geometries beforehand, even though it does not really matter what we put into the SRTree